api/queries_pr.go

```diff
@@ -109,10 +85,45 @@ type PullRequest struct {
 	ReviewRequests ReviewRequests
 }
 
+type PRRepository struct {
+	Name string
+}
+
 type Commit struct {
 	OID string `json:"oid"`
 }
 
+type PullRequestCommit struct {
+	Commit PullRequestCommitCommit
+}
+
+// PullRequestCommitCommit is like "Commit" but with StatusCheckRollup
+type PullRequestCommitCommit struct {
+	Oid               string
+	StatusCheckRollup struct {
+		Contexts struct {
+			Nodes []struct {
+				TypeName    string    `json:"__typename"`
+				Name        string    `json:"name"`
+				Context     string    `json:"context,omitempty"`
+				State       string    `json:"state,omitempty"`
+				Status      string    `json:"status"`
+				Conclusion  string    `json:"conclusion"`
+				StartedAt   time.Time `json:"startedAt"`
+				CompletedAt time.Time `json:"completedAt"`
+				DetailsURL  string    `json:"detailsUrl"`
+				TargetURL   string    `json:"targetUrl,omitempty"`
+			}
+		}
+	}
+}
+
+func (pr *PullRequest) StubCommit(oid string) {
```

Comment:
Looks like this is only used in `merge_test`, can this be done directly in the test?